src/docbuilder/rustwide_builder.rs

```diff
@@ -403,21 +407,29 @@ impl RustwideBuilder {
         // fill up disk space.
         // This also prevents having multiple builders using the same rustwide workspace,
         // which we don't do. Currently our separate builders use a separate rustwide workspace.
-        self.workspace
-            .purge_all_build_dirs()
-            .map_err(FailureError::compat)?;
+        {
+            let _span = info_span!("purge_all_build_dirs").entered();
+            self.workspace
+                .purge_all_build_dirs()
+                .map_err(FailureError::compat)?;
+        }
```

Comment:
```suggestion
        info_span!("purge_all_build_dirs").in_scope(|| {
            self.workspace
                .purge_all_build_dirs()
                .map_err(FailureError::compat)
                .map_err(Error::new)
        })?;
```

I'm not sure if it's worth it without `try` blocks to simplify the error handling, but there's `Span::in_scope` to do small bits of work rather than having to use block-scopes.

Reply:
I updated this occurrence. 

I can also imagine it being useful for rustwide itself to have tracing spans in some places, but seeing the progress on the `anyhow` PR I'm not too optimistic this will happen: https://github.com/rust-lang/rustwide/pull/76